aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -171,7 +193,18 @@ Tensor empty_like(const Tensor& self, const TensorOptions& options) {
                                        self.q_scale().toDouble(),
                                        self.q_zero_point().toLong());
   }
-  return at::empty(self.sizes(), options);
+
+  auto memory_format = optional_memory_format.value_or(MemoryFormat::Contiguous);
+  auto use_memory_format = memory_format;
+  if (memory_format == MemoryFormat::Preserve) {
+    if (self.is_contiguous(MemoryFormat::ChannelsLast)) {
+      use_memory_format = MemoryFormat::ChannelsLast;
+    } else {
+      use_memory_format = MemoryFormat::Contiguous;
```

Comment:
The fallback here seems hazardous. This means that when you add another channel format, it's a BC-breaking change to `preserve` calls. Because previously they would not have been preserved (fall through!) and then afterwards, they'd be detected and preserved.